src/cliffordsynthesis/encoding/SingleGateEncoder.cpp

```diff
@@ -119,6 +155,33 @@ LogicTerm SingleGateEncoder::createNoChange(
   return changes;
```

Comment:
Is the `SingleGateEncoder::createNoChange(
    const std::size_t pos, const std::size_t except,
    const std::optional<std::size_t> except2)` function here still necessary after the changes here? 
Maybe phrased differently: I believe this function is still used for two-qubit gates. Does a change similar to the change for single-qubit gates also make sense there? 
Actually, it shouldn't be necessary there at all, right? the new constraints completely cover the NoGate changes.

Reply:
It is not necessary anymore. The constraint that no gate should be applied to a qubit is independent of the specific gate anyway. So I think it makes more sense to formulate it independent of the single- and two-qubit constraints.